lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs

```diff
@@ -471,6 +494,104 @@ impl FilesystemBackend {
             },
         )))
     }
+
+    fn make_client_drive_query_information_response(
+        rdp_req: efs::ServerDriveQueryInformationRequest,
+        file: Option<&FileCacheObject>,
+    ) -> PduResult<RdpdrPdu> {
+        if file.is_none() {
+            return Ok(RdpdrPdu::ClientDriveQueryInformationResponse(
+                efs::ClientDriveQueryInformationResponse {
+                    device_io_response: efs::DeviceIoResponse::new(
+                        rdp_req.device_io_request.clone(),
+                        NtStatus::UNSUCCESSFUL,
+                    ),
+                    buffer: None,
+                },
+            ));
+        }
+        let file = file.unwrap(); // safe because of the if statement above
+        let device_io_response =
+            efs::DeviceIoResponse::new(rdp_req.device_io_request.clone(), NtStatus::SUCCESS);
+
+        // We support all the FsInformationClasses that FreeRDP does here
+        // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L482
+        match rdp_req.file_info_class_lvl {
```

Comment:
Can we extract branches below to separate methods? It's hard to follow

Reply:
This gets modified in https://github.com/gravitational/teleport/pull/34139:

https://github.com/gravitational/teleport/pull/34139/files#diff-1b8bd3bdf7e3fb4382d5b034f19725c768263110c7648f3199d74fc77558002fR517-R619

Moving this discussion there.

Reply:
a59a44ede95a22ba2fa1c848c78668c87cf1d71d